src/RentSplit/RentSplitState.kt

```diff
@@ -129,15 +154,45 @@ fun RentSplitState.addingNewExpense(newExpense: RentExpense): RentSplitState {
 /**
  * Converts this state into a string. First, all field names are shortened
  */
-fun RentSplitState.serialized(): String = JSON.stringify(this.toJson())
+fun RentSplitState.serialized(purpose: SerializationPurpose) = when (purpose) {
+    forLocalStorage -> JSON.stringify(this.toJson(purpose))
+    forSharing -> encodeURIComponent(JSON.stringify(this.toJson(purpose)))
```

Comment:
While this works to make the shared URI clickable by all tested interpreters, it also results in a very long URL. I'll create a new ticket (#29)